library/src/main/java/com/datatorrent/lib/projection/ProjectionOperator.java

```diff
@@ -0,0 +1,303 @@
+  protected transient List<TypeInfo> remainderFields = new ArrayList<>();
+
+  @AutoMetric
+  protected long projectedTuples;
+
+  @AutoMetric
+  protected long remainderTuples;
+
+  @AutoMetric
+  protected long errorTuples;
+
+  protected Class<?> inClazz = null;
+  protected Class<?> projectedClazz = null;
+  protected Class<?> remainderClazz = null;
+
+  @InputPortFieldAnnotation(schemaRequired = true)
+  public transient DefaultInputPort<Object> in = new DefaultInputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      inClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+
+    @Override
+    public void process(Object t)
+    {
+      handleProjection(t);
+    }
+  };
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> projected = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      projectedClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> remainder = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      remainderClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> error = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      inClazz = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+  /**
+   * addProjectedField: Add field details (name, type, getter and setter) for field with given name
+   * in projectedFields list
+   */
+  protected void addProjectedField(String s)
+  {
+    try {
+      Field f = inClazz.getDeclaredField(s);
+      TypeInfo t = new TypeInfo(f.getName(), ClassUtils.primitiveToWrapper(f.getType()));
+      t.getter = PojoUtils.createGetter(inClazz, t.name, t.type);
+      logger.debug("Creating setter {} {} {}", projectedClazz, t.name, t.type);
+      t.setter = PojoUtils.createSetter(projectedClazz, t.name, t.type);
+      projectedFields.add(t);
+    } catch (NoSuchFieldException e) {
+      throw new RuntimeException("Field " + s + " not found in class " + inClazz, e);
+    }
+  }
+
+  /**
+   * addRemainderField: Add field details (name, type, getter and setter) for field with given name
+   * in remainderFields list
+   */
+  protected void addRemainderField(String s)
+  {
+    try {
+      Field f = inClazz.getDeclaredField(s);
+      TypeInfo t = new TypeInfo(f.getName(), ClassUtils.primitiveToWrapper(f.getType()));
+      t.getter = PojoUtils.createGetter(inClazz, t.name, t.type);
+      t.setter = PojoUtils.createSetter(remainderClazz, t.name, t.type);
+      remainderFields.add(t);
+    } catch (NoSuchFieldException e) {
+      throw new RuntimeException("Field " + s + " not found in class " + inClazz, e);
+    }
+  }
+
```

Comment:
projectedFields and remainderFields are separate and extending class might be interested only in one of them. Hence populating these lists have been kept differently.